client/state/data-layer/wpcom/sites/activity/from-api.js

```diff
@@ -20,13 +20,15 @@ export const DEFAULT_GRIDICON = 'info-outline';
 /**
  * Transforms API response into array of activities
  *
- * @param  {object} apiResponse                      API response body
- * @param  {array}  apiResponse.current.orderedItems Array of item objects
- * @return {array}                                   Array of proccessed item objects
+ * @param  {object} apiResponse API response body
+ * @return {object}             Object with an entry for proccessed item objects and another for oldest item timestamp
  */
 export function transformer( apiResponse ) {
-	const orderedItems = get( apiResponse, [ 'current', 'orderedItems' ], [] );
-	return map( orderedItems, processItem );
+	return {
+		items: map( get( apiResponse, [ 'current', 'orderedItems' ], [] ), processItem ),
+		oldestItemTs: get( apiResponse, [ 'oldestItemTs' ], 0 ),
```

Comment:
as noted above, `0` isn't any kind of empty or meaningless value with dates. it's a middle value like 139,452. In this case it's `Jan 1, 1970` so maybe if it's missing we can leave it `null` or pick a value which our reducer should ignore like `Infinity` (If the "oldest" timestamp is infinitely in the future then when we compare new ones from the API against existing ones and pick the older of the two then this value will never be chosen).

Reply:
This is the 0 mentioned in https://github.com/Automattic/wp-calypso/pull/20216#discussion_r167583860